### DISCUSSION TITLE
How to subscribe on new posts in custom feed?

### DISCUSSION THREAD

**Discussion by lokxii** [MarshalX/atproto#307] [Questions] [answered]
[The specs](https://atproto.com/specs/event-stream) says that one can upgrade and HTTP connection to a websocket connection to receive event streams. However I can't find any examples / reference usage of such feature. Are there any plans for this SDK to support upgrading HTTP connection to websocket?

**Comment by MarshalX** [comment]
?? https://atproto.blue/en/latest/atproto_firehose/index.html

**Comment by lokxii** [comment]
This only subscribes to the global firehose. I am interpreting from that article that we can subscribe to events from a repo or from a list.

**Comment by MarshalX** [comment]
You can filter events by repo. And this will be stream by specific repo :)

**Comment by lokxii** [comment]
How about custom feeds?

**Comment by MarshalX** [comment]
@lokxii wdym? how to implement? check this https://github.com/MarshalX/bluesky-feed-generator

**Comment by lokxii** [comment]
I mean I thought I can websocket connect to a feed created by someone to listen to events. So I am imagining I can somehow connect to `wss://(that endpoint to get feed posts)`

**Comment by MarshalX** [comment, marked as answer]
@lokxii ah no. this is not how custom feed works :( the only way is to pull posts from custom feed using rest client https://docs.bsky.app/docs/tutorials/custom-feeds#viewing-a-feed-generators-posts

**Comment by lokxii** [comment]
Ohhh thx for your reply! I guess I will just spam get requests to keep getting posts.
